### PR TITLE
Service endpoint call read after create and update

### PR DESCRIPTION
**Pull request by xuzhang3**
* call read after create and update
* add state check during creation

## All Submissions:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] My code follows the code style of this project.
* [x] I ran lint checks locally prior to submission.
* [x] Have you checked to ensure there aren't other open PRs for the same update/change?

## What about the current behavior has changed?

PR:  #241 
Issue: #261 
1. Call read after create and update
2. Add state check during creation
3. Disable some service endpoint acceptance tests due to resource limit

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?

- [ ] Yes
- [x] No

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Any relevant logs, error output, etc?

(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

### CODE REVIEW

File: azuredevops/internal/acceptancetests/resource_serviceendpoint_kubernetes_test.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ const terraformServiceEndpointNode = "azuredevops_serviceendpoint_kubernetes.ser
 // validates that an apply followed by another apply (i.e., resource update) will be reflected in AzDO and the
 // underlying terraform state.
 func TestAccServiceEndpointKubernetesForAzureSubscriptionCreateAndUpdate(t *testing.T) {
+	t.Skip("Skipping test TestAccServiceEndpointKubernetesForAzureSubscriptionCreateAndUpdate: test resource limit")
```
Comment (EliiseS) [inline azuredevops/internal/acceptancetests/resource_serviceendpoint_kubernetes_test.go:24]: Is this a new issue?

Comment (xuzhang3) [reply azuredevops/internal/acceptancetests/resource_serviceendpoint_kubernetes_test.go:24]: I add state check during creation,   dummy value won't be able to create the real service connection though  we can find it in the `service connection` page.  I disable it due to test resource and sensitive configurations.

File: azuredevops/internal/service/serviceendpoint/commons.go [hunk 2]
```diff
@@ -169,8 +190,23 @@ func genServiceEndpointCreateFunc(flatFunc flatFunc, expandFunc expandFunc) func
 			return fmt.Errorf("Error creating service endpoint in Azure DevOps: %+v", err)
 		}
 
+		log.Printf("[DEBUG] Waiting service endpoint ready")
+		stateConf := &resource.StateChangeConf{
+			ContinuousTargetOccurence: 1,
+			Delay:                     10 * time.Second,
+			MinTimeout:                10 * time.Second,
+			Pending:                   []string{opState.Failed},
```
Comment (EliiseS) [inline azuredevops/internal/service/serviceendpoint/commons.go]: Shouldn't this say in progress? (in progress is not used)

File: azuredevops/internal/service/serviceendpoint/commons.go [hunk 3]
```diff
@@ -240,3 +276,29 @@ func genServiceEndpointDeleteFunc(expandFunc expandFunc) schema.DeleteFunc {
...
+		if err != nil {
+			return nil, opState.Failed, fmt.Errorf("Error looking up service endpoint given ID (%v) and project ID (%v): %v ", serviceEndpointID, projectID, err)
+		}
+
+		if *serviceEndpoint.IsReady {
+			return serviceEndpoint, opState.Ready, nil
+		} else if serviceEndpoint.OperationStatus != nil {
+			opStatus := (serviceEndpoint.OperationStatus).(map[string]interface{})
+			if opStatus["state"] == opState.Failed {
+				return nil, opState.Failed, fmt.Errorf("Error looking up service endpoint given ID (%v) and project ID (%v): %v ", serviceEndpointID, projectID, serviceEndpoint.OperationStatus)
```
Comment (EliiseS) [inline azuredevops/internal/service/serviceendpoint/commons.go]: This error message is the same as on line 291, should this text be placed in a variable?

**Review by EliiseS — APPROVED** [review]

**Review by EliiseS — APPROVED** [review]

**Comment by xuzhang3** [comment]
\

**Pull request merged**